cilium/cmd/root.go

```diff
@@ -155,6 +167,16 @@ func runCompletion(out io.Writer, cmd *cobra.Command, args []string) error {
 	if _, err := out.Write([]byte(copyRightHeader)); err != nil {
 		return err
 	}
+	if len(args) == 0 {
+		return cmd.Parent().GenBashCompletion(out)
+	}
+
+	switch args[0] {
+	case "bash":
+		return cmd.Parent().GenBashCompletion(out)
+	case "zsh":
+		return cmd.Parent().GenZshCompletion(out)
+	}
 
-	return cmd.Parent().GenBashCompletion(out)
+	return fmt.Errorf("Unexpected shell: %s.", args[0])
```

Comment:
Just follow the error format in this function. Maybe we could change them all.

```go
func runCompletion(out io.Writer, cmd *cobra.Command, args []string) error {
	if len(args) > 1 {
		return fmt.Errorf("Too many arguments. Expected only the shell type.") // <--------
	}
	if _, err := out.Write([]byte(copyRightHeader)); err != nil {
		return err
	}
	if len(args) == 0 {
		return cmd.Parent().GenBashCompletion(out)
	}

	switch args[0] {
	case "bash":
		return cmd.Parent().GenBashCompletion(out)
	case "zsh":
		return cmd.Parent().GenZshCompletion(out)
	}

	return fmt.Errorf("Unexpected shell: %s.", args[0])
}
```

Reply:
Yes, feel free to fix the other one too.
Note that the error is actually unused. Would you mind changing the call as follows so that errors are actually taken into account?
```go
RunE: func(cmd *cobra.Command, args []string) error {
    return runCompletion(out, cmd, args)
},
```

Reply:
Yep, I am going to change Run to RunE, and return the error msg to outer.